lib/ansible/plugins/action/copy.py

```diff
@@ -290,8 +290,7 @@ def _copy_file(self, source_full, source_rel, content, content_tempfile,
 
             if self._play_context.check_mode:
                 self._remove_tempfile_if_content_defined(content, content_tempfile)
-                result['changed'] = True
-                return result
+                raise AnsibleActionSkip('Check mode is not supported for this task.')
```

Comment:
Isn't this just removing check mode support? The only time skipping makes sense to me (sorta; I think 'skipped' is overloaded) is if changed is unknown (such as for `script`), and here we know changed=True.